The First Prototype

AndrewDPham opened the pull request:
Users can login and take a quiz. There is a student quiz for them to answer and a teacher quiz for the admin to edit. There is also a sign out button. For now, we need to fix the account on the iFrame (doesn't match the login status of the html page). We also removed the API keys and created a "test files" folder which holds our test code. Lastly, servlets are not functional so no need to check them out.

kyleasmith1 committed:
Initial commit

kyleasmith1 committed:
Update README.md

kyleasmith1 committed:
Creation...

AndrewDPham committed:
Login works

AndrewDPham committed:
Start of second page

kyleasmith1 committed:
Remove target and rename portfolio

AndrewDPham committed:
Merge branch 'master' of https://github.com/googleinterns/step152-2020 into login

AndrewDPham committed:
Merge branch 'master' of https://github.com/googleinterns/step152-2020 into login

kyleasmith1 committed:
Sync Andrew and I's repos

kyleasmith1 committed:
Sync branch

kyleasmith1 committed:
Apps

AndrewDPham committed:
Change index.html to send a token, create auth servlet

AndrewDPham committed:
Add Java dependencies, JS CallTestScript

kyleasmith1 committed:
update to index.html

kyleasmith1 committed:
Merge branch 'apps' of https://github.com/googleinterns/step152-2020 into apps

AndrewDPham committed:
Fail debug with Scott

kyleasmith1 committed:
Rename files and organize

AndrewDPham committed:
Change scope

AndrewDPham committed:
Merge branch 'prototype' of https://github.com/googleinterns/step152-2020 into prototype

kyleasmith1 committed:
Updates to the login page

kyleasmith1 committed:
Merge branch 'prototype-A' of https://github.com/googleinterns/step152-2020 into prototype-K

AndrewDPham committed:
Add form.html

kyleasmith1 committed:
Merge branch 'prototype-A' of https://github.com/googleinterns/step152-2020 into prototype-K

kyleasmith1 committed:
organize and set up prototype

kyleasmith1 committed:
remove api keys and sensitive data

AndrewDPham committed:
Refactor code, remove trivial lines

kyleasmith1 commented:
@googlebot I fixed it.

AndrewDPham commented:
@googlebot I fixed it.

kyleasmith1 committed:
Merge branch 'protoReal' of https://github.com/googleinterns/step152-2020 into protoReal

707090 commented on capstone/src/main/java/com/google/sps/servlets/AuthenticateServlet.java:
```diff
@@ -0,0 +1,56 @@
+        response.getWriter().println("<h1>Hello world!</h1>");
+        response.getWriter().println(request);
+
+    //     GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(transport, jsonFactory)
+    //     // Specify the CLIENT_ID of the app that accesses the backend:
+    //     .setAudience(Collections.singletonList(CLIENT_ID))
+    //     // Or, if multiple clients access the backend:
+    //     //.setAudience(Arrays.asList(CLIENT_ID_1, CLIENT_ID_2, CLIENT_ID_3))
+    //     .build();
+
+    //     // (Receive idTokenString by HTTPS POST)
+
+    //     GoogleIdToken idToken = verifier.verify(idTokenString);
+    //     if (idToken != null) {
+    //     Payload payload = idToken.getPayload();
+
+    //     // Print user identifier
+    //     String userId = payload.getSubject();
+    //     System.out.println("User ID: " + userId);
+
+    // // Get profile information from payload
+    //     String email = payload.getEmail();
+    //     boolean emailVerified = Boolean.valueOf(payload.getEmailVerified());
+    //     String name = (String) payload.get("name");
+    //     String pictureUrl = (String) payload.get("picture");
+    //     String locale = (String) payload.get("locale");
+    //     String familyName = (String) payload.get("family_name");
+    //     String givenName = (String) payload.get("given_name");
+
+    // // Use or store profile information
+    // // ...
+    //     } else {
+    //         System.out.println("Invalid ID token.");
```
Code style wise, PRs should not have any commented code, debug code, or mis-formatted code. (applies everywhere)

AndrewDPham replied:
See below.

707090 commented on capstone/src/main/webapp/form.html:
```diff
@@ -0,0 +1,20 @@
+    <div id="content">
+      <p><h1>Quiz</h1></p>
+      <iframe src="https://docs.google.com/forms/d/e/1FAIpQLScrpUcq5Jnv6knV6K3aexWa1m972EmMzHJUvXulHCfBsJaEpw/viewform?embedded=true">Loading…</iframe>
+      <br>
+      <br>
```
If the goal is some whitespace, CSS should handle this. Use HTML tags when it reflects the actual structure of the data, and CSS when it is purely a visual thing

AndrewDPham replied:
Alright, sounds good. We will delete the html tags that deal with style for now and will focus on styling the page in a future PR.

707090 commented on capstone/src/main/webapp/index.html:
```diff
@@ -0,0 +1,22 @@
+      <div id="content">
+        <p><h1>Home</h1></p>
+        <a href="form.html">Take A Quiz!</a>
+        &nbsp;&nbsp;
```
Same as above, CSS over HTML

AndrewDPham replied:
Sounds good. Same as previous comment.

707090 commented on capstone/src/main/webapp/script.js:
```diff
@@ -0,0 +1,39 @@
+  console.log(profile.getEmail() + " has signed in!");
+}
+
+function myInfo(){
```
Choosing informative names helps navigate the code, so since this only prints things, it should be named something like "printOauthProfile" or something.

AndrewDPham replied:
Sounds good. We deleted this function in future branches but for now, we can have it for master.

707090 commented on capstone/src/main/webapp/test files/apps.html:
```diff
@@ -0,0 +1,32 @@
+        gapi.client.init({
+          'apiKey': '',
+          'discoveryDocs': [''],
+        }).then(function() {
+          // Executes an API request, and returns a Promise.
+          // The method name `language.translations.list` comes from the API discovery.
+          return gapi.client.language.translations.list({
+            q: 'hello world',
+            source: 'en',
+            target: 'de',
+          });
+        }).then(function(response) {
+          console.log(response.result.data.translations[0].translatedText);
+        }, function(reason) {
+          console.log('Error: ' + reason.result.error.message);
+        });
+      };
```
This looks like it... translates "hello world" to dutch? Is there a reason this is here?

AndrewDPham replied:
See below.

707090 commented on capstone/src/main/webapp/test files/home.html:
```diff
@@ -0,0 +1,31 @@
+    </head>
+    <body>
+    <div id="content">
+        <div class="g-signin2" data-onsuccess="onSignIn"></div>
```
Is there a reason the class is named "xxx-2"?

AndrewDPham replied:
See below.

707090 commented on capstone/src/main/webapp/test files/scripts.html:
```diff
@@ -0,0 +1,127 @@
+        gapi.auth2.getAuthInstance().signOut();
+      }
+
+      /**
+       * Append a pre element to the body containing the given message
+       * as its text node. Used to display the results of the API call.
+       *
+       * @param {string} message Text to be placed in pre element.
+       */
+      function appendPre(message) {
+        var pre = document.getElementById('content');
+        var textContent = document.createTextNode(message + '\n');
+        pre.appendChild(textContent);
+      }
```
Is this used?

AndrewDPham replied:
See below.

707090 commented on capstone/src/main/webapp/test files/scripts.html:
```diff
@@ -0,0 +1,127 @@
+       * Call the Apps Script API to create a new script project, upload files
+       * to the project, and log the script's URL to the user.
+       */
+
```
whitespace

AndrewDPham replied:
See below.

707090 commented on capstone/src/main/webapp/test files/test.html:
```diff
@@ -0,0 +1,208 @@
+      /**
+       *  Called when the signed in status changes, to update the UI
+       *  appropriately. After a sign-in, the API is called.
+       */
+      function updateSigninStatus(isSignedIn) {
+        if (isSignedIn) {
+          authorizeButton.style.display = 'none';
+          signoutButton.style.display = 'block';
+          callAppsScript();
+        } else {
+          authorizeButton.style.display = 'block';
+          signoutButton.style.display = 'none';
+        }
+      }
+
+      /**
+       *  Sign in the user upon button click.
+       */
+      function handleAuthClick(event) {
+        gapi.auth2.getAuthInstance().signIn();
+      }
+
+      /**
+       *  Sign out the user upon button click.
+       */
+      function handleSignoutClick(event) {
+        gapi.auth2.getAuthInstance().signOut();
+      }
+
+      /**
+       * Append a pre element to the body containing the given message
+       * as its text node. Used to display the results of the API call.
+       *
+       * @param {string} message Text to be placed in pre element.
+       */
+      function appendPre(message) {
+        var pre = document.getElementById('content');
+        var textContent = document.createTextNode(message + '\n');
+        pre.appendChild(textContent);
+      }
```
Repeated code should be pulled out into a separate js file and pulled in with a script tag in the head

AndrewDPham replied:
See below.

AndrewDPham committed:
Delete unnecessary files

AndrewDPham commented:
@707090 We deleted the "test files" since they are not needed and purely for testing and we also deleted the servlets since we did not use them. Sorry about that, Kyle and I will keep the PR "ready" to merge into master from now on.

AndrewDPham committed:
Fix styling on HTML, renamed myInfo() function in script.js

kyleasmith1 committed:
Merge branch 'protoReal' of https://github.com/googleinterns/step152-2020 into protoReal

kyleasmith1 committed:
Add User.java

AndrewDPham committed:
Remove p tag

AndrewDPham committed:
Merge branch 'protoReal' of https://github.com/googleinterns/step152-2020 into protoReal
